### PR TITLE
Sync master with constraint support 

### PR DESCRIPTION
**Pull request by lnash94**

### CODE REVIEW

File: openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/ModuleMemberVisitor.java [hunk 1]
```diff
@@ -32,8 +31,8 @@
  */
 public class ModuleMemberVisitor extends NodeVisitor {
 
-    LinkedList<TypeDefinitionNode> typeDefinitionNodes = new LinkedList<>();
-    LinkedList<ListenerDeclarationNode> listenerDeclarationNodes = new LinkedList<>();
+    LinkedHashSet<TypeDefinitionNode> typeDefinitionNodes = new LinkedHashSet<>();
+    LinkedHashSet<ListenerDeclarationNode> listenerDeclarationNodes = new LinkedHashSet<>();
```
Comment (ayeshLK) [inline openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/ModuleMemberVisitor.java:35]:
```suggestion
    Set<TypeDefinitionNode> typeDefinitionNodes = new LinkedHashSet<>();
    Set<ListenerDeclarationNode> listenerDeclarationNodes = new LinkedHashSet<>();
```

File: openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/ModuleMemberVisitor.java [hunk 2]
```diff
@@ -45,11 +44,11 @@ public void visit(ListenerDeclarationNode listenerDeclarationNode) {
         listenerDeclarationNodes.add(listenerDeclarationNode);
     }
 
-    public List<TypeDefinitionNode> getTypeDefinitionNodes() {
+    public LinkedHashSet<TypeDefinitionNode> getTypeDefinitionNodes() {
```
Comment (ayeshLK) [inline openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/ModuleMemberVisitor.java:47]:
```suggestion
    public Set<TypeDefinitionNode> getTypeDefinitionNodes() {
```

File: openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/ModuleMemberVisitor.java [hunk 3]
```diff
@@ -45,11 +44,11 @@ public void visit(ListenerDeclarationNode listenerDeclarationNode) {
...
         return typeDefinitionNodes;
     }
 
-    public List<ListenerDeclarationNode> getListenerDeclarationNodes() {
+    public LinkedHashSet<ListenerDeclarationNode> getListenerDeclarationNodes() {
```
Comment (ayeshLK) [inline openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/ModuleMemberVisitor.java:51]:
```suggestion
    public Set<ListenerDeclarationNode> getListenerDeclarationNodes() {
```

File: openapi-bal-service/src/main/java/io/ballerina/openapi/converter/diagnostic/DiagnosticMessages.java [hunk 4]
```diff
@@ -61,12 +61,19 @@ public enum DiagnosticMessages {
             DiagnosticSeverity.WARNING),
     OAS_CONVERTOR_114("OAS_CONVERTOR_114", "Generated OpenAPI definition does not contain information " +
             "for Ballerina type '%s'. ", DiagnosticSeverity.WARNING),
-
     OAS_CONVERTOR_115("OAS_CONVERTOR_115", "Given Ballerina file does not contain any HTTP service.",
             DiagnosticSeverity.ERROR),
-    OAS_CONVERTOR_116("OAS_CONVERTOR_116", "Generated OpenAPI definition does not contain `%s` request" +
+    OAS_CONVERTOR_116("OAS_CONVERTOR_116", "Failed to parser the Number value due to: %s ",
```
Comment (dilanSachi) [inline openapi-bal-service/src/main/java/io/ballerina/openapi/converter/diagnostic/DiagnosticMessages.java:66]:
```suggestion
    OAS_CONVERTOR_116("OAS_CONVERTOR_116", "Failed to parse the Number value due to: %s ",
```

Comment (dilanSachi) [reply openapi-bal-service/src/main/java/io/ballerina/openapi/converter/diagnostic/DiagnosticMessages.java:66]: Should be parse right?

File: openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/ConstraintAnnotation.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com).
```
Comment (dilanSachi) [inline openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/ConstraintAnnotation.java:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
```
Is this a file created in 2022 or a mistake?

File: openapi-cli/src/test/resources/ballerina-to-openapi/constraint/array.bal [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    PersonFeeItemsNumber[] fee?;
+    # The maximum number of items in the response (as set in the query or by default).
+    PersonLimitItemsInteger[] 'limit?;
+};
+
+service /payloadV on new http:Listener(9090) {
+    resource function post pet(@http:Payload Person body) returns error? {
+        return;
+    }
+}
```
Comment (dilanSachi) [inline openapi-cli/src/test/resources/ballerina-to-openapi/constraint/array.bal:40]: newline

File: openapi-cli/src/test/java/io/ballerina/openapi/generators/openapi/ConstraintTests.java [hunk 7]
```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com).
```
Comment (dilanSachi) [inline openapi-cli/src/test/java/io/ballerina/openapi/generators/openapi/ConstraintTests.java:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
```
here also

**Pull request merged**